### ISSUE TITLE
Could not find a valid Docker environment

### ISSUE THREAD

**Issue by YikSanChan** [testcontainers/testcontainers-scala#175]
Hi,

I have a minimal test.

```scala
import com.dimafeng.testcontainers.{
  Container,
  ForAllTestContainer,
  GenericContainer
}
import org.scalatest.matchers.should
import org.scalatest.wordspec.AnyWordSpec

class MyTest extends AnyWordSpec with should.Matchers with ForAllTestContainer {
  override val container: Container =
    GenericContainer("redis:latest", exposedPorts = Seq(6379))

  "Test" should {
    "pass" in {
      1 shouldBe 1
    }
  }
}
```

When I run it, it throws exception:

```
Exception encountered when invoking run on a nested suite - Could not find a valid Docker environment. Please see logs and check configuration
java.lang.IllegalStateException: Could not find a valid Docker environment. Please see logs and check configuration
	at org.testcontainers.dockerclient.DockerClientProviderStrategy.lambda$getFirstValidStrategy$7(DockerClientProviderStrategy.java:215)
	at java.base/java.util.Optional.orElseThrow(Optional.java:401)
	at org.testcontainers.dockerclient.DockerClientProviderStrategy.getFirstValidStrategy(DockerClientProviderStrategy.java:207)
	at org.testcontainers.DockerClientFactory.getOrInitializeStrategy(DockerClientFactory.java:136)
	at org.testcontainers.DockerClientFactory.client(DockerClientFactory.java:178)
	at org.testcontainers.LazyDockerClient.getDockerClient(LazyDockerClient.java:14)
	at org.testcontainers.LazyDockerClient.authConfig(LazyDockerClient.java:12)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:310)
	at com.dimafeng.testcontainers.SingleContainer.start(Container.scala:46)
	at com.dimafeng.testcontainers.ForAllTestContainer$class.run(ForAllTestContainer.scala:17)
	at flink.redis.MyTest.run(MyTest.scala:11)
	at org.scalatest.tools.SuiteRunner.run(SuiteRunner.scala:45)
	at org.scalatest.tools.Runner$$anonfun$doRunRunRunDaDoRunRun$1.apply(Runner.scala:1322)
	at org.scalatest.tools.Runner$$anonfun$doRunRunRunDaDoRunRun$1.apply(Runner.scala:1316)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at org.scalatest.tools.Runner$.doRunRunRunDaDoRunRun(Runner.scala:1316)
	at org.scalatest.tools.Runner$$anonfun$runOptionallyWithPassFailReporter$2.apply(Runner.scala:972)
	at org.scalatest.tools.Runner$$anonfun$runOptionallyWithPassFailReporter$2.apply(Runner.scala:971)
	at org.scalatest.tools.Runner$.withClassLoaderAndDispatchReporter(Runner.scala:1482)
	at org.scalatest.tools.Runner$.runOptionallyWithPassFailReporter(Runner.scala:971)
	at org.scalatest.tools.Runner$.run(Runner.scala:798)
	at org.scalatest.tools.Runner.run(Runner.scala)
	at org.jetbrains.plugins.scala.testingSupport.scalaTest.ScalaTestRunner.runScalaTest2or3(ScalaTestRunner.java:38)
	at org.jetbrains.plugins.scala.testingSupport.scalaTest.ScalaTestRunner.main(ScalaTestRunner.java:25)
```

My pom includes:

```
		<dependency>
			<groupId>com.dimafeng</groupId>
			<artifactId>testcontainers-scala-scalatest_2.11</artifactId>
			<version>0.39.3</version>
			<scope>test</scope>
		</dependency>
```

What is going on? Thanks

**Comment by dimafeng** [comment]
It seems that the library can't access the docker client on your machine.

**Comment by YikSanChan** [comment]
@dimafeng I wonder why is that the case? In terminal I am able to run `docker ps` and it lists the containers running.

**Comment by dimafeng** [comment]
Can you share more details about your system. Does this https://github.com/testcontainers/testcontainers-java/issues/3834 look like your case?

**Comment by YikSanChan** [comment]
@dimafeng I am not using Apple Silicon.

Here's my docker info.

```
> docker version
Client: Docker Engine - Community
 Cloud integration: 1.0.9
 Version:           20.10.5
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        55c4c88
 Built:             Tue Mar  2 20:13:00 2021
 OS/Arch:           darwin/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.5
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       363e9a8
  Built:            Tue Mar  2 20:15:47 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Comment by fsanaulla** [comment]
Same here. Take a look on my CICD [build](https://github.com/fsanaulla/spark-http-rdd/runs/2390257531?check_suite_focus=true)

**Comment by fsanaulla** [comment]
Btw, it was fixed after refacroting

**Comment by YikSanChan** [comment]
@fsanaulla What kind of refactoring you did?

**Comment by fsanaulla** [comment]
@YikSanChan everything was done in this [commit](https://github.com/fsanaulla/spark-http-rdd/commit/4f4d6220b3547f478331465040be04b1225189b7).
Long story short:

- moving away from docker compose to single containers
- more precise dependencies
- prebuild image

**Comment by roodnejm** [comment]
Hi, if it can help, i had the same issue as mentioned in the first post if i used a version > 0.38.5 on macOs. 
With 0.38.5, I had another error about testcontainers/ryuk:0.3.0 image not found (solution is to used latest version of testcontainer according to a testcontainer-java issue  but not possible in my case due to the "could not find a valid Docker environment" error.

So in the end, I used a dockercompose testcontainer (localstack + postgresql) with version 0.38.5 and seems to work.

**Comment by seuf** [comment]
Hello,
I have the same issue when I try to run my tests in gitlab-ci with docker:dind service.
I have exported the `DOCKER_HOST=tcp://docker:2376` environment variable, but test container ping is still trying to access `/var/run/docker.sock` (NoSuchFileException).
I'm using the same method on other gitlab-ci pipelines with Java testcontainers and it works.

Here is my logs..

```
WARN TestcontainersConfiguration: Attempted to read Testcontainers configuration file at file:/home/sbtuser/.testcontainers.properties but the file was not found. Exception message: FileNotFoundException: /home/sbtuser/.testcontainers.properties (No such file or directory)
INFO ImageNameSubstitutor: Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
INFO DockerMachineClientProviderStrategy: docker-machine executable was not found on PATH ([/usr/local/openjdk-11/bin, /usr/local/sbin, /usr/local/bin, /usr/sbin, /usr/bin, /sbin, /bin])
ERROR DockerClientProviderStrategy: Could not find a valid Docker environment. Please check configuration. Attempted configurations were:
ERROR DockerClientProviderStrategy:     EnvironmentAndSystemPropertyClientProviderStrategy: failed with exception TimeoutException (Timeout waiting for result with exception). Root cause ConnectException (Connection refused (Connection refused))
ERROR DockerClientProviderStrategy:     UnixSocketClientProviderStrategy: failed with exception InvalidConfigurationException (Could not find unix domain socket). Root cause NoSuchFileException (/var/run/docker.sock)
ERROR DockerClientProviderStrategy: As no valid configuration was found, execution cannot continue
```

I've also tried to configure `.testcontainers.properties`  according to [test container documentation](https://www.testcontainers.org/features/configuration/#customizing-docker-host-detection) with the following conf:
```
docker.client.strategy=org.testcontainers.dockerclient.EnvironmentAndSystemPropertyClientProviderStrategy
docker.host=tcp\://docker\:2376
docker.tls.verify
docker.tls.certdir=/certs
docker.cert.path=/certs/client
docker.driver=overlay2
```

But it still looking for /var/run/docker.sock :disappointed:

**Comment by Yummy-Yums** [comment]
@dimafeng is there a soln for this issue  i'm running into this after trying to run the below project

SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
  - order repository test with postgres test container
    Exception in thread "zio-fiber-22" zio.test.TestFailure$Runtime: Runtime(Fail(java.lang.IllegalStateException: Could not find a valid Docker environment. Please see logs and check configuration,Stack trace for thread "zio-fiber-":
    ),Map())
    	at sviezypan.repo.OrderRepositoryLiveSpec.spec(OrderRepositoryLiveSpec.scala:37)
    	at sviezypan.repo.OrderRepositoryLiveSpec.spec(OrderRepositoryLiveSpec.scala:37)
    	at sviezypan.repo.OrderRepositoryLiveSpec.spec(OrderRepositoryLiveSpec.scala:37)
0 tests passed. 0 tests failed. 0 tests ignored.

https://github.com/sviezypan/zio-sql-example

**Comment by dimafeng** [comment]
I suspect it's not related to this library but rather the configuration of your local docker environment. You can also search for solutions in the parent library repo https://java.testcontainers.org/

**Comment by Yummy-Yums** [comment]
If you mean adding Docker environment variables, I did that, I also tried running postgres as standalone rather than use the compose up

I'd try checking but is there any resource for setting testcontainers up for Scala or Java should suffice?

**Comment by Winand** [comment]
Update. The problem is https://github.com/testcontainers/testcontainers-java/issues/11212#issuecomment-3516810057

To resolve this I've set _test/resources/docker-java.properties:_  `api.version=1.45`

---

I get error for npipe on Windows 10, Scala 2.11, testcontainers 0.41.3.
I've even tried to call it directly:
```scala
  val x = new NpipeSocketClientProviderStrategy
  val cmd = x.getDockerClient.infoCmd
  val ex = cmd.exec
```

```
com.github.dockerjava.api.exception.BadRequestException: Status 400: 
{
    "ID": "",
    "Containers": 0,
    "ContainersRunning": 0,
    "ContainersPaused": 0,
    "ContainersStopped": 0,
    "Images": 0,
    "Driver": "",
    "DriverStatus": null,
    "Plugins": {
        "Volume": null,
        "Network": null,
        "Authorization": null,
        "Log": null
    },
    "MemoryLimit": false,
    "SwapLimit": false,
    "CpuCfsPeriod": false,
    "CpuCfsQuota": false,
    "CPUShares": false,
    "CPUSet": false,
    "PidsLimit": false,
    "IPv4Forwarding": false,
    "Debug": false,
    "NFd": 0,
    "OomKillDisable": false,
    "NGoroutines": 0,
    "SystemTime": "",
    "LoggingDriver": "",
    "CgroupDriver": "",
    "NEventsListener": 0,
    "KernelVersion": "",
    "OperatingSystem": "",
    "OSVersion": "",
    "OSType": "",
    "Architecture": "",
    "IndexServerAddress": "",
    "RegistryConfig": null,
    "NCPU": 0,
    "MemTotal": 0,
    "GenericResources": null,
    "DockerRootDir": "",
    "HttpProxy": "",
    "HttpsProxy": "",
    "NoProxy": "",
    "Name": "",
    "Labels": ["com.docker.desktop.address=npipe://\\\\.\\pipe\\docker_cli"],
    "ExperimentalBuild": false,
    "ServerVersion": "",
    "Runtimes": null,
    "DefaultRuntime": "",
    "Swarm": {
        "NodeID": "",
        "NodeAddr": "",
        "LocalNodeState": "",
        "ControlAvailable": false,
        "Error": "",
        "RemoteManagers": null
    },
    "LiveRestoreEnabled": false,
    "Isolation": "",
    "InitBinary": "",
    "ContainerdCommit": {
        "ID": ""
    },
    "RuncCommit": {
        "ID": ""
    },
    "InitCommit": {
        "ID": ""
    },
    "SecurityOptions": null,
    "CDISpecDirs": null,
    "Warnings": null
}
```
<details>
<summary>Debug messages:</summary>
```
23:04:02.266 [ScalaTest-run] DEBUG org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd -- Cmd: 
23:04:02.297 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.InternalHttpClient -- ex-00000001: preparing request execution
23:04:02.304 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.protocol.RequestAddCookies -- Cookie spec selected: strict
23:04:02.310 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.protocol.RequestAuthCache -- Auth cache not set in the context
23:04:02.310 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ProtocolExec -- ex-00000001: target auth state: UNCHALLENGED
23:04:02.311 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ProtocolExec -- ex-00000001: proxy auth state: UNCHALLENGED
23:04:02.311 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ConnectExec -- ex-00000001: acquiring connection with route {}->npipe://localhost:2375
23:04:02.311 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.InternalHttpClient -- ex-00000001: acquiring endpoint (3 MINUTES)
23:04:02.313 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager -- ex-00000001: endpoint lease request (3 MINUTES) [route: {}->npipe://localhost:2375][total available: 0; route allocated: 0 of 2147483647; total allocated: 0 of 2147483647]
23:04:02.319 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager -- ex-00000001: endpoint leased [route: {}->npipe://localhost:2375][total available: 0; route allocated: 1 of 2147483647; total allocated: 1 of 2147483647]
23:04:02.336 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager -- ex-00000001: acquired ep-00000000
23:04:02.336 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.InternalHttpClient -- ex-00000001: acquired endpoint ep-00000000
23:04:02.336 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ConnectExec -- ex-00000001: opening connection {}->npipe://localhost:2375
23:04:02.337 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.InternalHttpClient -- ep-00000000: connecting endpoint (3 MINUTES)
23:04:02.337 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager -- ep-00000000: connecting endpoint to npipe://localhost:2375 (3 MINUTES)
23:04:02.343 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.DefaultHttpClientConnectionOperator -- http-outgoing-0: connecting to localhost/127.0.0.1:2375
23:04:02.350 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.DefaultHttpClientConnectionOperator -- http-outgoing-0: connection established 
23:04:02.350 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager -- ep-00000000: connected http-outgoing-0
23:04:02.350 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.InternalHttpClient -- ep-00000000: endpoint connected
23:04:02.350 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.MainClientExec -- ex-00000001: executing GET /v1.32/info HTTP/1.1
23:04:02.351 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.InternalHttpClient -- ep-00000000: start execution ex-00000001
23:04:02.351 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager -- ep-00000000: executing exchange ex-00000001 over http-outgoing-0
23:04:02.352 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers -- http-outgoing-0 >> GET /v1.32/info HTTP/1.1
23:04:02.352 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers -- http-outgoing-0 >> x-tc-sid: 6f0dbd66-8ad2-49bf-8503-ceca0501a3f4
23:04:02.352 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers -- http-outgoing-0 >> User-Agent: tc-java/1.19.1
23:04:02.352 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers -- http-outgoing-0 >> Accept-Encoding: gzip, x-gzip, deflate
23:04:02.352 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers -- http-outgoing-0 >> Host: localhost:2375
23:04:02.352 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers -- http-outgoing-0 >> Connection: keep-alive
23:04:02.352 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 >> "GET /v1.32/info HTTP/1.1[\r][\n]"
23:04:02.352 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 >> "x-tc-sid: 6f0dbd66-8ad2-49bf-8503-ceca0501a3f4[\r][\n]"
23:04:02.352 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 >> "User-Agent: tc-java/1.19.1[\r][\n]"
23:04:02.352 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 >> "Accept-Encoding: gzip, x-gzip, deflate[\r][\n]"
23:04:02.352 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 >> "Host: localhost:2375[\r][\n]"
23:04:02.352 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 >> "Connection: keep-alive[\r][\n]"
23:04:02.352 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 >> "[\r][\n]"
23:04:02.358 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 << "HTTP/1.1 400 Bad Request[\r][\n]"
23:04:02.358 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 << "Api-Version: 1.52[\r][\n]"
23:04:02.358 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 << "Content-Type: application/json[\r][\n]"
23:04:02.358 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 << "Date: Mon, 29 Dec 2025 20:04:02 GMT[\r][\n]"
23:04:02.358 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 << "Ostype: linux[\r][\n]"
23:04:02.358 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 << "Server: Docker/29.1.2 (linux)[\r][\n]"
23:04:02.358 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 << "Transfer-Encoding: chunked[\r][\n]"
23:04:02.358 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 << "[\r][\n]"
23:04:02.358 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 << "4b0[\r][\n]"
23:04:02.358 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 << "{"ID":"","Containers":0,"ContainersRunning":0,"ContainersPaused":0,"ContainersStopped":0,"Images":0,"Driver":"","DriverStatus":null,"Plugins":{"Volume":null,"Network":null,"Authorization":null,"Log":null},"MemoryLimit":false,"SwapLimit":false,"CpuCfsPeriod":false,"CpuCfsQuota":false,"CPUShares":false,"CPUSet":false,"PidsLimit":false,"IPv4Forwarding":false,"Debug":false,"NFd":0,"OomKillDisable":false,"NGoroutines":0,"SystemTime":"","LoggingDriver":"","CgroupDriver":"","NEventsListener":0,"KernelVersion":"","OperatingSystem":"","OSVersion":"","OSType":"","Architecture":"","IndexServerAddress":"","RegistryConfig":null,"NCPU":0,"MemTotal":0,"GenericResources":null,"DockerRootDir":"","HttpProxy":"","HttpsProxy":"","NoProxy":"","Name":"","Labels":["com.docker.desktop.address=npipe://\\\\.\\pipe\\docker_cli"],"ExperimentalBuild":false,"ServerVersion":"","Runtimes":null,"DefaultRuntime":"","Swarm":{"NodeID":"","NodeAddr":"","LocalNodeState":"","ControlAvailable":false,"Error":"","RemoteManagers":null},"LiveRestoreEnabled":false,"Isolation":"","InitBinary":"","ContainerdCommit":{"ID":""},"RuncCommit":{"ID":""},"InitCommit":{"ID":""},"SecurityOptions":null,"CDISpecDirs":null,"Warnings":null}[\r][\n]"
23:04:02.358 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 << "0[\r][\n]"
23:04:02.358 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire -- http-outgoing-0 << "[\r][\n]"
23:04:02.361 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers -- http-outgoing-0 << HTTP/1.1 400 Bad Request
23:04:02.361 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers -- http-outgoing-0 << Api-Version: 1.52
23:04:02.361 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers -- http-outgoing-0 << Content-Type: application/json
23:04:02.361 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers -- http-outgoing-0 << Date: Mon, 29 Dec 2025 20:04:02 GMT
23:04:02.361 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers -- http-outgoing-0 << Ostype: linux
23:04:02.361 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers -- http-outgoing-0 << Server: Docker/29.1.2 (linux)
23:04:02.361 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers -- http-outgoing-0 << Transfer-Encoding: chunked
23:04:02.367 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.MainClientExec -- ex-00000001: connection can be kept alive for 3 MINUTES
23:04:02.376 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.InternalHttpClient -- ep-00000000: releasing valid endpoint
23:04:02.376 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager -- ep-00000000: releasing endpoint
23:04:02.376 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager -- ep-00000000: connection http-outgoing-0 can be kept alive for 3 MINUTES
23:04:02.376 [ScalaTest-run] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager -- ep-00000000: connection released [route: {}->npipe://localhost:2375][total available: 1; route allocated: 1 of 2147483647; total allocated: 1 of 2147483647]
```
</details>
